app/models/service.rb

```diff
@@ -69,55 +70,28 @@ def request_type
     'service_reconfigure'
   end
 
-  def root_service
-    result = self
-    until result.parent_service.nil?
-      result = result.parent_service
-    end
-    result
-  end
-
-  def ancestors
-    result = []
-    node = self
-
-    while (node = node.parent_service)
-      result << node
-    end
-
-    result
-  end
-
-  def direct_service_children
-    services
-  end
+  alias root_service root
+  alias services children
+  alias direct_service_children children
 
   def indirect_service_children
-    direct_service_children.collect { |s| s.direct_service_children + s.indirect_service_children }.flatten.compact
-  end
-
-  def descendants
-    all_service_children
-  end
-
-  def subtree
-    all_service_children + [self]
+    descendants(:from_depth => 2)
   end
+  Vmdb::Deprecation.deprecate_methods(self, :indirect_service_children)
```

Comment:
most (all?) of the code that calls indirect is doing this to get the `subtree`.
kept it around because it was exposed to automate